### ISSUE TITLE
Sublibrary for encrypted libraries

### ISSUE THREAD

**Issue by devillemereuil** [haiwen/seafile#1029]
Hi!

I noticed that "Sublibraries for encrypted libraries" is gone from the Roadmap. Does that mean the feature has been dropped? I can easily see why, but since I was counting on this feature in the future, I wanted a confirmation from you guys that I need to change my plans (not sure how...).

Thank you for the good work!!

**Comment by freeplant** [comment]
We are going to implement syncing of any sub-folder from the client side (selective sync). But it will be pro edition only.

For non pro edition user, they can use the old "sub-library" way. Sub library for encrypted libraries may be added too. But we haven't review the tech part carefully. It may not secure to enable sub-libraries for encrypted libraries.

**Comment by devillemereuil** [comment]
Thank you for the response!

When you say "sublibraries for for encrypted libraries may be added too", you mean that there is a way for me, the administrator, to do it, or that you guys may implement it in a future version?

**Comment by shoeper** [comment]
```
We are going to implement syncing of any sub-folder from the client side (selective sync). But it will be pro edition only.
```
That's not nice to hear. I think that's absolutely basic functionality.

**Comment by devillemereuil** [comment]
Hi @freeplant 

I'm sorry to insist, but could you tell if there is a way to active Sub-libraries for encrypted libraries on my server? It seems the feature is working, because old Sub-libraries for old encrypted libraries are still available on my server. Also, my alternative is using Dropbox to share my files with other, which is not that secure anyway...

Thank you for any help you could provide on that.

**Comment by freeplant** [comment]
We will add sub-libraries for encrypted libraries.

**Comment by devillemereuil** [comment]
OK, thank you for that. Sorry if I have been oppressive on you, wasn't the goal.

**Comment by freeplant** [comment]
I see you need to share a sub-folder of an encrypted library to another user. Will you tell the password of the whole library to the other?

**Comment by devillemereuil** [comment]
Yes, that is my need. This is because I have a rather large library that cannot be separated into many libraries (due to my working flow and structure), and I need to share part of it with other people.

And yes, my plan was to securely share the password with a few others (that I trust sufficiently to create an account for them on my Seafile server). Actually, since passwords of libraries can now be changed, I feel this is more secure than it was before because the password can be changed as soon as one fears it has been compromised (this is always an issue when sharing passwords with a third party).

Do you fear this is too big of a security issue?

**Comment by Bran1** [comment]
Has this been implemented for the pro edition only?

**Comment by freeplant** [comment]
Sub folder share for encrypted libraries is not implemented yet.

**Comment by freeplant** [comment]
Here is the latest states after so many years:

1. Syncing of any sub-folder for non encrypted libraries is implemented  in both CE and PRO editions.
2. Syncing of any sub-folder for encrypted libraries are not implemented due to technique limitation.
